Text/Shakespeare/Base.hs

```diff
@@ -177,6 +186,12 @@ derefToExp s (DerefTuple ds) = TupE $
                                map Just $
 #endif
                                map (derefToExp s) ds
+derefToExp s (DerefGetField x f) =
+#if MIN_VERSION_template_haskell(2,18,0)
+    GetFieldE (derefToExp s x) f
+#else
+    error "Your compiler doesn't support OverloadedRecordDot"
+#endif
```

Comment:
So, supporting older `TemplateHaskell` would mean putting in an `Exp` that corresponded to `getField` - 

That seems cool to me! Though, I'm not sure if I'd want to block merging this on that - if you're using `OverloadedRecordDot`, presumably you are on a new enough GHC and TH to warrant it.

Reply:
> if you're using OverloadedRecordDot, presumably you are on a new enough GHC and TH to warrant it.

Yeah, that's my thinking as well.

I think it's better to use `GetFieldE` and throw an error on compilers that don't support `OverloadedRecordDot`.  The `getField` desugaring means we have to reach into `GHC.Records`, and hope that API remains stable.